src/main/java/com/maddyhome/idea/vim/group/OptionGroup.kt

```diff
@@ -8,124 +8,33 @@
 
 package com.maddyhome.idea.vim.group
 
-import com.intellij.openapi.application.ApplicationNamesInfo
 import com.maddyhome.idea.vim.api.VimOptionGroupBase
-import com.maddyhome.idea.vim.options.OptionConstants
-import com.maddyhome.idea.vim.options.StringOption
-import com.maddyhome.idea.vim.options.ToggleOption
-import com.maddyhome.idea.vim.options.UnsignedNumberOption
 
 internal class OptionGroup : VimOptionGroupBase() {
-
   init {
-    addOption(ToggleOption(IjOptionConstants.closenotebooks, IjOptionConstants.closenotebooks, true))
-    addOption(
-      StringOption(
-        IjOptionConstants.ide,
-        IjOptionConstants.ide,
-        ApplicationNamesInfo.getInstance().fullProductNameWithEdition,
-      ),
-    )
-    addOption(ToggleOption(IjOptionConstants.ideacopypreprocess, IjOptionConstants.ideacopypreprocess, false))
-    addOption(ToggleOption(IjOptionConstants.ideajoin, IjOptionConstants.ideajoin, false))
-    addOption(ToggleOption(IjOptionConstants.ideamarks, IjOptionConstants.ideamarks, true))
-    addOption(
-      StringOption(
-        IjOptionConstants.idearefactormode,
-        IjOptionConstants.idearefactormode,
-        "select",
-        isList = false,
-        IjOptionConstants.ideaRefactorModeValues,
-      ),
-    )
-    addOption(
-      StringOption(
-        IjOptionConstants.ideastatusicon,
-        IjOptionConstants.ideastatusicon,
-        "enabled",
-        isList = false,
-        IjOptionConstants.ideaStatusIconValues,
-      ),
-    )
-    addOption(
-      StringOption(
-        IjOptionConstants.ideavimsupport,
-        IjOptionConstants.ideavimsupport,
-        "dialog",
-        isList = true,
-        IjOptionConstants.ideavimsupportValues,
-      ),
-    )
-    addOption(
-      StringOption(
-        IjOptionConstants.ideawrite,
-        IjOptionConstants.ideawrite,
-        "all",
-        isList = false,
-        IjOptionConstants.ideaWriteValues,
-      ),
-    )
-    addOption(
-      StringOption(
-        IjOptionConstants.lookupkeys,
-        IjOptionConstants.lookupkeys,
-        "<Tab>,<Down>,<Up>,<Enter>,<Left>,<Right>,<C-Down>,<C-Up>,<PageUp>,<PageDown>,<C-J>,<C-Q>",
-        isList = true,
-      ),
-    )
-    addOption(ToggleOption(IjOptionConstants.oldundo, IjOptionConstants.oldundo, true))
-    addOption(ToggleOption(IjOptionConstants.trackactionids, "tai", false))
-    addOption(UnsignedNumberOption(IjOptionConstants.visualdelay, IjOptionConstants.visualdelay, 100))
-
-    // This options overrides Vim's default value, so we keep it here
-    addOption(
-      StringOption(
-        OptionConstants.clipboard,
-        OptionConstants.clipboardAlias,
-        "ideaput,autoselect,exclude:cons\\|linux",
-        isList = true,
-      ),
-    )
-
-    addOption(ToggleOption("unifyjumps", "unifyjumps", true))
+    IjOptions.initialise()
```

Comment:
Is there a chance that `Options.clipboard` will be accessed before the `OptionGroup` is initialized?

Reply:
This is trickier than I realised. The idea of storing the option definitions inside `VimOptionGroup` would create a circular dependency - when `Options` is initialised, it would call into `VimOptionGroup` which might need to initialise and which calls `IjOptions.initialise`, which adds option definitions via `Options`, which would try to call into `VimOptionGroup` again.

We could make `Option.defaultValue` internal, so no-one could access it except via `VimOptionGroup`, but internal just means internal to vim-engine, so it doesn't offer much protection.

I think the next best alternative is for `Options.init` to notify _someone_ that options have been initialised, at which point, _someone_ could override `Options.clipboard` (and probably set up the IntelliJ specific options too). I'm not sure who _someone_ is though. It might be an implementation interface for `VimOptionGroup`, or even a new service that does nothing in vim-engine, but can be overridden in the IntelliJ build. What do you think?

Reply:
Or, we take `ideaput` out of the defaults and have a notification like we do for `ideajoin`. The first time we put without `ideaput` set, show a notification, offer to modify `.ideavimrc` and set a flag to not show the notification again.